src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -375,6 +370,8 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
     } else if (status.getState().equals(TaskState.TASK_RUNNING)) {
         stagingTasks.remove(status.getTaskId());
         clusterState.updateTask(status);
+        List<TaskID> currentStagingTasksList = new ArrayList<TaskID>();
+        currentStagingTasksList.addAll(stagingTasks);
```

Comment:
Why do you even need `currentStagingTasksList` anymore? Just iterate directly over `stagingTasks`